### ISSUE TITLE
data.ibm_cos_bucket does not work correctly in version 1.44.0

### ISSUE THREAD

**Issue by rich-ehrhardt** [IBM-Cloud/terraform-provider-ibm#3944]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform IBM Provider Version

Terraform v1.2.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/external v2.2.2
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.3.2
+ provider registry.terraform.io/ibm-cloud/ibm v1.44.0

### Affected Resource(s)

data.ibm_cos_bucket

### Terraform Configuration Files

```hcl
provider "ibm" {
  ibmcloud_api_key = var.ibmcloud_api_key
  region = var.region
}

terraform {
    required_providers {
        ibm = {
            source = "ibm-cloud/ibm"
            version = "1.44.0"
        }
    }
}

variable "ibmcloud_api_key" {
  type = string
  description = "the value of ibmcloud_api_key"
}

variable "region" {
  type = string
  description = "the value of region"
}

module "cos-resource_group" {
  source = "cloud-native-toolkit/resource-group/ibm"
  version = "3.3.2"

  ibmcloud_api_key = var.ibmcloud_api_key
  purge_volumes = true
  resource_group_name = "rbe-test-rg"
  sync = ""
}

data "ibm_resource_group" "cos_group" {
    depends_on = [
      module.cos-resource_group
    ]
    name = module.cos-resource_group.name
}

resource "ibm_resource_instance" "cos_instance" {
    name = "rbe-cos-instance"
    resource_group_id = data.ibm_resource_group.cos_group.id
    service = "cloud-object-storage"
    plan = "standard"
    location = "global"
}

resource "ibm_cos_bucket" "cos_test_bucket" {
    bucket_name = "rbe-cos-test-bucket"
    resource_instance_id = ibm_resource_instance.cos_instance.id
    region_location = "eu-gb"
    storage_class = "standard"
}

data "ibm_cos_bucket" "cos_test_bucket" {
    depends_on = [
      resource.ibm_cos_bucket.cos_test_bucket
    ]

    bucket_name = "rbe-cos-test-bucket"
    resource_instance_id = ibm_resource_instance.cos_instance.id
    bucket_type = "single_site_location"
    bucket_region = "eu-gb"
}

output "cos_bucket_crn" {
    value = data.ibm_cos_bucket.cos_test_bucket.crn
}
```

### Debug Output

data.ibm_cos_bucket.cos_test_bucket: Reading...
2022-08-03T17:57:41.319+1000 [ERROR] vertex "data.ibm_cos_bucket.cos_test_bucket" error: ReplicationConfigurationNotFoundError: The replication configuration was not found
        status code: 404, request id: a557d446-4444-46aa-946e-8b9a43c971dd, host id:
╷
│ Error: ReplicationConfigurationNotFoundError: The replication configuration was not found
│       status code: 404, request id: a557d446-4444-46aa-946e-8b9a43c971dd, host id: 
│ 
│   with data.ibm_cos_bucket.cos_test_bucket,
│   on main.tf line 57, in data "ibm_cos_bucket" "cos_test_bucket":
│   57: data "ibm_cos_bucket" "cos_test_bucket" {

### Expected Behavior

The data.ibm_cos_bucket module should successfully read the details of the COS bucket that has been created.

### Actual Behavior

As per debug output above, the data.ibm_cos_bucket module reports a configuration error related to the replication configuration even though no replication has been set for the bucket.

### Steps to Reproduce

1. `terraform apply --auto-approve`

### Important Factoids

The same code above works with version 1.43.0

### References

* #0000

**Comment by hkantare** [comment]
We are looking into the issue

**Comment by hkantare** [comment]
Fixed in latest release 1.44.1
https://github.com/IBM-Cloud/terraform-provider-ibm/releases/tag/v1.44.1

**Comment by rich-ehrhardt** [comment]
Tested with updated version 1.44.1 and problem is fixed.
